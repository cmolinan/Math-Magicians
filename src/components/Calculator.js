import React from 'react';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="AllCalculator">
        <input type="text" id="input" />
        <div className="allButtons">
          <div className="row">
            <button type="button" className="btnWhite" id="AC">AC</button>
            <button type="button" className="btnWhite" id="sign">+/-</button>
            <button type="button" className="btnWhite" id="percent">%</button>
            <button type="button" className="btnOrange" id="div">/</button>
          </div>
          <div className="row">
            <button type="button" className="btnWhite" id="b7">7</button>
            <button type="button" className="btnWhite" id="b8">8</button>
            <button type="button" className="btnWhite" id="b9">9</button>
            <button type="button" className="btnOrange" id="mult">x</button>
          </div>
          <div className="row">
            <button type="button" className="btnWhite" id="b4">4</button>
            <button type="button" className="btnWhite" id="b5">5</button>
            <button type="button" className="btnWhite" id="b6">6</button>
            <button type="button" className="btnOrange" id="sub">-</button>
          </div>
          <div className="row">
            <button type="button" className="btnWhite" id="b1">1</button>
            <button type="button" className="btnWhite" id="b2">2</button>
            <button type="button" className="btnWhite" id="b3">3</button>
            <button type="button" className="btnOrange" id="sum">+</button>
          </div>
          <div className="lastrow">
            <button type="button" className="btnWhite" id="b0">0</button>
            <button type="button" className="btnWhite" id="point">.</button>
            <button type="button" className="btnOrange" id="equal">=</button>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
