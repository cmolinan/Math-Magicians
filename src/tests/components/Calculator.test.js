import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../../components/Calculator';

const DomContainer = () => {
  <BrowserRouter>
    <Calculator />
  </BrowserRouter>;
};

it('Check if renders', () => {
  const tree = renderer.create(<DomContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check if a value is displayed', () => {
  render(<Calculator />);
  const buttonPressed = screen.getByText(/9/i);
  fireEvent.click(buttonPressed);

  const displayElement = document.getElementById('input');

  expect(displayElement.value).toBe('9');
});

it('Check if it is displayed the result of an operation', () => {
  render(<Calculator />);
  const firstNum = screen.getByText(/4/i);
  const opKey = screen.getByText('x');
  const secondNum = screen.getByText(/9/i);
  const eqKey = screen.getByText('=');

  const displayElement = document.getElementById('input');

  fireEvent.click(firstNum);
  fireEvent.click(opKey);
  fireEvent.click(secondNum);
  fireEvent.click(eqKey);
  expect(displayElement.value).toBe('36');
});
