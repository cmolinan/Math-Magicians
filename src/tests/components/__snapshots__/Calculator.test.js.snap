// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if renders 1`] = `null`;
